src/emissions/emission.py

```diff
@@ -44,46 +64,50 @@ def __init__(
         trajectory : Trajectory
             Flight trajectory for mission object with altitude, speed,
             and fuel mass time series.
-        EDB_data : bool
-            Flag indicating whether to use EDB tabulated data (True)
-            or user specified LTO data (False).
-        fuel_file : str
-            Path to TOML file containing fuel properties
         """
 
-        # Load fuel properties from TOML
-        fuel_file_loc = file_location(ac_performance.config['fuel_file'])
-        with open(fuel_file_loc, 'rb') as f:
+        self.trajectory = trajectory
+        self.performance_model = ac_performance
+        settings = self._parse_emission_settings(ac_performance.config)
+        self.settings = settings
+        self.traj_emissions_all = settings.traj_all
+        self.pmvol_method = settings.pmvol_method
+        self.pmnvol_method = settings.pmnvol_method
+        self.apu_enabled = settings.apu_enabled
+        self.gse_enabled = settings.gse_enabled
+        self.lifecycle_enabled = settings.lifecycle_enabled
+        self.metric_flags = dict(settings.metric_flags)
+        self.method_flags = dict(settings.method_flags)
+        self._include_pmnvol_number = self.metric_flags[
+            'PMnvol'
+        ] and self.pmnvol_method in ('scope11', 'meem')
+        self._scope11_cache = None
+
+        with open(file_location(settings.fuel_file), 'rb') as f:
             self.fuel = tomllib.load(f)
+        self.co2_ei, self.nvol_carb_cont = EI_CO2(self.fuel)
```

Comment:
Loose data types: don't use anonymous tuples like this. Be more explicit about what types values can have. "Type" means more than "int or float or np.ndarray": a language like Python gives you lots of ways to make your own types that are more meaningful, whether that's using normal classes or dataclasses or something else.